lib/src/block_parser.dart

```diff
@@ -144,6 +143,20 @@ abstract class BlockSyntax {
     if (parser.isDone) return true;
     return parser.blockSyntaxes.any((s) => s.canParse(parser) && s.canEndBlock);
   }
+
+  /// Generate a valid HTML anchor from the inner text of [element].
+  static String generateAnchorHash(Element element) =>
+      _concatenatedText(element)
+          .toLowerCase()
+          .replaceAll(new RegExp(r'^[^a-z]+'), '')
+          .replaceAll(new RegExp(r'[^a-z0-9 _-]'), '')
+          .replaceAll(new RegExp(r'\s'), '-')
+          .replaceAll(new RegExp(r'-$'), '');
```

Comment:
What about leading whitespace? Might be simplest just to trim the string before all the replaces.